pkg/controllers/dynakube/processmoduleconfigsecret/reconciler.go

```diff
@@ -143,6 +171,22 @@ func (r *Reconciler) prepareSecret(ctx context.Context) (*corev1.Secret, error)
 	return newSecret, err
 }
 
+func (r *Reconciler) getAgentTenantToken(ctx context.Context) (string, error) {
```

Comment:
it's duplicated [GetDataFromSecretName](https://github.com/Dynatrace/dynatrace-operator/blob/main/pkg/util/kubeobjects/secret/secret.go#L180) function. Error message wrapper could be "secret not found or invalid" and details comes from the err returned by the function

Reply:
Done in https://github.com/Dynatrace/dynatrace-operator/pull/2777/commits/3ac9f4a406e5def836a71c7ddb0c336baee14ffc